addons/wowl/static/src/legacy/legacy.ts

```diff
@@ -55,30 +43,39 @@ odoo.define("wowl.legacySetup", async function (require: any) {
       });
     },
   };
-  serviceRegistry.add(legacyRpcService.name, legacyRpcService);
+}
 
-  let webClientReady = new Promise((resolve) => {
-    const busToOwlLegacyBus: Service<void> = {
-      name: "bus_to_owl_legacy_bus",
-      deploy(env: OdooEnv): void {
-        env.bus.on("WEB_CLIENT_READY", null, resolve);
-      },
-    };
-    serviceRegistry.add(busToOwlLegacyBus.name, busToOwlLegacyBus);
+export function mapLegacyEnvToWowlEnv(legacyEnv: any, wowlEnv: OdooEnv) {
+  // rpc
+  legacyEnv.session.rpc = (...args: any[]) => {
+    let rejection;
+    const prom = new Promise((resolve, reject) => {
+      rejection = () => reject();
+      const [route, params, settings] = args;
+      wowlEnv.services.rpc(route, params, settings).then(resolve).catch(reject);
+    });
+    (prom as any).abort = rejection;
+    return prom;
+  };
+  // localStorage
+  const localStorage = wowlEnv.browser.localStorage;
+  legacyEnv.services.local_storage = Object.assign(Object.create(localStorage), {
+    getItem(key: string, defaultValue: any) {
```

Comment:
need a localStorage Service to:
- handle crash in Safari
- modify getItem/setItem API to JSON stringify/parse

browser should be equal to window stuff